js/script.js

```diff
@@ -0,0 +1,10 @@
+function startTime() {
+  var today = new Date();
+  var h = today.getHours().toString().padStart(2, '0');
+  var m = today.getMinutes().toString().padStart(2, '0');
+  var s = today.getSeconds().toString().padStart(2, '0');
+  document.getElementById('time').innerHTML = `${h}:${m}:${s}`;
+  window.setTimeout(() => startTime(), 500);
```

Comment:
Dangerous code. You're calling function from itself - it may grep nearest context closure which will eat your memory.
Take a look at timeline JSheap record of your page (90sec record with no actions from user):
![](https://i.imgur.com/K5LZdu6.png)

Heap grows during 1 min, and only then GC cleaning memory.

I think its better to do just:
```js
(function() {
  var time = document.getElementById('time');
  function startTime() {
    var today = new Date();
    var h = today.getHours().toString().padStart(2, '0');
    var m = today.getMinutes().toString().padStart(2, '0');
    var s = today.getSeconds().toString().padStart(2, '0');
    time.innerHTML = `${h}:${m}:${s}`;
    
  }

  startTime();
  window.setTimeout(() => startTime(), 500);
})()
```
Also pay attention at time block - its better to get HTML once instead of hitting DOM every 0.5sec